content/client-lib-development-guide/features.textile

```diff
@@ -1534,40 +1478,59 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  + getDeviceInfo() => io DeviceDetails
+  id: String
+  platform: DeviceType
+  type: DeviceType
+  clientId: String?
+  metadata: Dict<String, String>
+  updateTokenId: String
+  resetId(String)
+  resetUpdateToken() => io
```

Comment:
I am assuming therefore there is no way to configure the `updateTokenId`?  I appreciate most of the time this will be unnecessary, however if the `updateTokenId` was invalid for some reason, then I assume `resetUpdateToken` would not work because it cannot authenticate, or would it instead rely on the client being authenticated to obtain a new `updateTokenId`.  If the latter, then this in itself is a bit of a security concern as our servers would then have no way to confirm if the request for a new token is valid or not for that `device.ID` and as such you could in theory take control of someone else's `device.ID`.  Perhaps the thinking is as follows:

* `updateTokenId` is generated by Ably when the device is first registered, and returned in the registration response
* An endpoint will exist that allows any device to generate a new `updateTokenId` and can do this when authenticating with the existing `updateTokenId`
* A client library can reset it's `updateTokenId` by calling `resetUpdateToken()` which handles everything transparently if the existing `updateTokenId` is valid.
* A client library can explicitly reset it's `updateTokenId` by calling `resetUpdateTokenId(string)`, but the token string can only be obtained server-side by a client with `push-admin` rights i.e. that endpoint to reset the `updateTokenId` is callable with that privilege

Note that I think a lot of this functionality is unnecessary for MVP launch, and at launch, we should probably just focus on ensuring that when registered, a device obtains and stores a valid `updateTokenId` only, and we can worry about `resetId, resetUpdateToken, resetUpdateTokenId` post MVP launch.  Not saying we don't define it now in the IDL though...

On a separate note, I think `resetId` should require a new `updateTokenId` too as the old `updateTokenId` will no longer be valid.  So perhaps then given the above, we only have two methods:

`reset(deviceId, updateTokenId)` and `renewUpdateToken() => io`

WDYT?